lib/rubocop/cop/rspec/each.rb

```diff
@@ -0,0 +1,53 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Check that all matcher is used instead of iterating over an array.
+      #
+      # @example
+      #   # bad
+      #   it 'validates users' do
+      #     [user1, user2, user3].each { |user| expect(user).to be_valid }
+      #   end
+      #
+      #   # good
+      #   it 'validates users' do
+      #     expect([user1, user2, user3]).to all(be_valid)
+      #   end
+      class Each < Cop
+        MSG = 'Prefer using the `all` matcher instead ' \
+                  'of iterating over an array.'.freeze
+
+        def_node_matcher :each?, <<-PATTERN
+          (block
+            (send ... :each)
+            (args (arg $_))
+            $(...)
+          )
+        PATTERN
+
+        def_node_matcher :expectation?, <<-PATTERN
+          (send (send nil :expect (lvar %)) :to ...)
+        PATTERN
+
+        def on_block(node)
+          each?(node) do |arg, body|
+            if single_expectation?(body, arg) || only_expectations?(body, arg)
+              @arg = arg
```

Comment:
left over from some initial attempts to implement autocorrect. Its not needed, as arg can be retrieved from the node itself, but I have dropped the autocorrect support completely. I'll clean up